examples/md-flexible/scripts/generate-mdflex-configs.py

```diff
@@ -143,34 +167,52 @@ def generate(domainSize,
                    .replace('Lennard-Jones 12-6', 'LJ')
                    .replace(' ', '_'))
 
+    # This writes all flow style values (lists enclosed in []) to block style (list of -).
+    # While they are not fully equivalent they are mostly interchangeable and not a problem in our context.
     with open(outFilename, 'w') as fileOutput:
         yaml.dump(data, fileOutput, sort_keys=False)
 
 
-def isInteresting(domainSize,
-                  numParticles,
-                  distribution,
-                  cutoff,
-                  skinFactor,
+def isInteresting(domainSizeName,
+                  numParticlesName,
+                  distributionName,
+                  cutoffName,
+                  skinFactorName,
                   invMaxParticleSpeed,
                   functor,
-                  cellSizeFactor):
+                  cellSizeFactorName):
 
     """
     Filter to judge if a config is worth to generate.
     """
 
-    tooDenseThreshold = 400  # particles per cell
-    actualDomainSize = domainSizes[domainSize]
-    numCells = actualDomainSize[0] * actualDomainSize[1] * actualDomainSize[2]
-    uniformAvgParticlesPerCell = particleCounts[numParticles] / numCells
-
-    return not (uniformAvgParticlesPerCell > tooDenseThreshold
-                or (numParticles == 'very-few' and functor == 'lj-avx')
-                or (numParticles == 'huge' and distribution != 'uniform-whole')
-                # or (cellSizeFactor == 'big')
-                # or (cutoff == 'big')
-                # or (functor == 'lj-no-avx')
+    cellSizeFactor = cellSizeFactors[cellSizeFactorName]
+    domainSize = domainSizes[domainSizeName]
+    cellSize = cutoffs[cutoffName] * (1 + skinFactors[skinFactorName]) * cellSizeFactor
+    cellsPerDimension = [math.floor(d / cellSize) for d in domainSize]
+    # If cells are larger than the domain in any dimension the scenario doesn't make sense
+    if any(item == 0 for item in cellsPerDimension):
+        return False
+    numCells = math.prod(cellsPerDimension)
+    uniformAvgParticlesPerCell = particleCounts[numParticlesName] / numCells
+    # Currently unused:
+    # volume=math.prod(domainSize)
+    # density=particleCounts[numParticlesName]/volume
+
+    # Exclusion rules. Mostly to avoid scenarios that take forever to test.
```

Comment:
This is a slightly awkward one. These parameters would exclude some of the current test cases which I am working on. But whilst I could suggest values which are more appropriate for my current test cases, these values could easily become as inappropriate in the future as these are for my test cases. My suggestion is to leave these as they are but add the warning:

```suggestion
    # Exclusion rules. Mostly to avoid scenarios that take forever to test. Warning: These parameters may not be appropriate for your test cases.
```